### PR TITLE
feat: authenticate with connected apps

### PR DESCRIPTION
**Pull request by JuanCBasso**
Port of fixed created in https://github.com/avioconsulting/mule-deploy-library/commit/5f3052aa09809c96e1407a26ee630503b56cec07 into this project around the HTTP Wrapper, as it's using the same structure.

### CODE REVIEW

File: src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package com.avioconsulting.mule.designcenter
 
+import com.avioconsulting.mule.designcenter.api.models.credentials.ConnectedAppCredential
+import com.avioconsulting.mule.designcenter.api.models.credentials.Credential
+import com.avioconsulting.mule.designcenter.api.models.credentials.UsernamePasswordCredential
```
Comment (manikmagar) [inline src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy:5]: Why `designcenter` package? This model applies to the authentication of anypoint platform. Maybe we should use `accessmanagement` or just `anypoint`?

Comment (JuanCBasso) [reply src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy:5]: It's a very valid observation @manikmagar, I do think it should go to `anypoint` too now that you mention it.

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/ConnectedAppCredential.groovy
Comment (manikmagar) [inline bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/ConnectedAppCredential.groovy:1]: I guess this `bin` directory should not be committed.

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/Credential.groovy
Comment (manikmagar) [inline bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/Credential.groovy:1]: I guess this `bin` directory should not be committed.

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/UsernamePasswordCredential.groovy
Comment (manikmagar) [inline bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/UsernamePasswordCredential.groovy:1]: I guess this `bin` directory should not be committed.

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy
Comment (manikmagar) [inline bin/src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy:1]: I guess this `bin` directory should not be committed.

**Review by manikmagar — CHANGES_REQUESTED** [review]
Please Remove the `bin` directory.

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/ConnectedAppCredential.groovy
Comment (JuanCBasso) [reply bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/ConnectedAppCredential.groovy:1]: Agree, removed and added bin folder into gitIgnore file

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/Credential.groovy
Comment (JuanCBasso) [reply bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/Credential.groovy:1]: Agree, removed and added bin folder into gitIgnore file

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/UsernamePasswordCredential.groovy
Comment (JuanCBasso) [reply bin/src/main/groovy/com/avioconsulting/mule/designcenter/api/model/credentials/UsernamePasswordCredential.groovy:1]: Agree, removed and added bin folder into gitIgnore file

File: bin/src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy
Comment (JuanCBasso) [reply bin/src/main/groovy/com/avioconsulting/mule/designcenter/HttpClientWrapper.groovy:1]: Agree, removed and added bin folder into gitIgnore file

**Review by manikmagar — APPROVED** [review]

**Pull request merged**